setup.py

```diff
@@ -73,40 +73,15 @@ def run(self):
 if platform.machine() == 'ppc64le':
     extra_link_args += ['-pthread']
 
-build_ext = cmdclass.get('build_ext', build_ext)
-
-numba_be_user_options = [
-    ('werror', None, 'Build extensions with -Werror'),
-    ('wall', None, 'Build extensions with -Wall'),
-    ('noopt', None, 'Build extensions without optimization'),
-]
-
 
 class NumbaBuildExt(build_ext):
-
-    user_options = build_ext.user_options + numba_be_user_options
-    boolean_options = build_ext.boolean_options + ['werror', 'wall', 'noopt']
-
-    def initialize_options(self):
-        super().initialize_options()
-        self.werror = 0
-        self.wall = 0
-        self.noopt = 0
-
     def run(self):
         extra_compile_args = []
-        if self.noopt:
-            if sys.platform == 'win32':
-                extra_compile_args.append('/Od')
-            else:
-                extra_compile_args.append('-O0')
-        if self.werror:
-            extra_compile_args.append('-Werror')
-        if self.wall:
-            extra_compile_args.append('-Wall')
+        if sys.platform.startswith("linux") and platform.machine() == "x86_64":
```

Comment:
I think this is not the behaviour we want - we don't want these flags to always be added on Linux / x86_64 when we build, only on Linux / x86_64 in CI. This still removes the option for the user to add these flags through options to `build_ext` (and additionally the `noopt` flag as well..